packages/firebase_analytics/firebase_analytics_platform_interface/lib/src/event_parameters.dart

```diff
@@ -0,0 +1,43 @@
+const String _exceptionMessage =
+    "'string' OR 'number' must be set as the value of the parameter";
+
+class EventParameters {
+  EventParameters();
+  // Constructs an [EventParameters] from a raw Map.
+  factory EventParameters.fromMap(Map<String, Object> map) {
+    Map<String, Object> parameters = {};
+    map.forEach((key, value) {
+      assert(
+        value is String || value is num,
+        _exceptionMessage,
+      );
+      parameters[key] = value;
+    });
+    return EventParameters._(parameters);
+  }
+
+  EventParameters._(this._parameters);
+
+  Map<String, Object> _parameters = {};
+
+  // Can only pass either a String or a num value
+  EventParameters addParameter(String key, {String? string, num? number}) {
```

Comment:
IMO, this is not much better. We are allowing the use to pass both of them, and he's going to have a runtime exception if he does it. We could make a safe API that wouldn't allow the user to pass any kind of invalid data, so that, if it compiles, it works.

Reply:
Fair point. I think I will change `addParameter` to `addString` & `addNumber` to improve it. 

I will be keeping the `fromMap()` function, as its main purpose is to help transition to the new `EventParameters` API, and it will still throw assert on incorrect property values before it ever makes it to the native side.